src/subscription_manager/identity.py

```diff
@@ -60,13 +65,37 @@ def exists(cls):
 
     @classmethod
     def existsAndValid(cls):
-        if cls.exists():
-            try:
-                cls.read()
-                return True
-            except Exception, e:
-                log.warn('possible certificate corruption')
-                log.error(e)
+        return cls.exists() and cls.valid()
+
+    @classmethod
+    def accessible(cls):
+        if not os.access(cls.certpath(), os.R_OK):
+            log.warn('Insufficient permissions to access cert file: %s' % cls.certpath())
+            return False
+        if not os.access(cls.keypath(), os.R_OK):
+            log.warn('Insufficient permissions to access key file: %s' % cls.keypath())
+            return False
+        return True
+
+    @classmethod
+    def valid(cls):
+        try:
+            cls.read()
+            return True
+        except CertificateException as e:
+            log.warn('possible certificate corruption')
+            log.error(e)
+            # Leave it to the caller to handle the exception
+            # so that they have to opportunity to display it nicely
+            # to the user (if appropriate)
+        except IOError as e:
+            log.warn('One of the consumer certificates could not be read')
+            log.error(e)
```

Comment:
I'm guessing we want to combine these into something like:
```
log.exception('One of the consumer certificates could not be read')
```

so that we get the full stack trace and cut down on logging clutter?